apps/src/weblab2/CDOIDE/cdoIDEContext/utils.ts

```diff
@@ -27,11 +27,20 @@ import {
 
 const DEFAULT_NEW_FILE_CONTENTS = 'Add your changes to ${fileName}';
 
-export const getNextFileId = (files: ProjectFile[]) =>
-  String(Math.max(...files.map(f => Number(f.id))) + 1);
+export const getNextFileId = (files: ProjectFile[]) => {
+  if (files.length === 0) {
+    return '1';
+  }
+  return String(Math.max(...files.map(f => Number(f.id))) + 1);
+};
 
-export const getNextFolderId = (folders: ProjectFolder[]) =>
-  String(Math.max(...folders.map(f => Number(f.id))) + 1);
+export const getNextFolderId = (folders: ProjectFolder[]) => {
+  if (folders.length === 0) {
+    return '1';
+  } else {
+    return String(Math.max(...folders.map(f => Number(f.id))) + 1);
+  }
+};
```

Comment:
Good catch!

I'd suggest fixing it like this:
```
export const getNextFolderId = (folders: ProjectFolder[]) =>
  String(Math.max(0, ...folders.map(f => Number(f.id))) + 1);
```

That way we don't need the conditional, but we'll still start at "1" if the list is empty.

Reply:
Nice that's much cleaner